lua/gatekeeper/policylib.lua

```diff
@@ -281,3 +281,36 @@ function decision_web(policy, tx_rate_kib_sec, cap_expire_sec,
 		policy, tx_rate_kib_sec, tx_rate_kib_sec * 0.05, -- 5%
 		cap_expire_sec, next_renewal_ms, renewal_step_ms, false)
 end
+
+local function lpm_add_tbl8s(ip_addr, prefix_len, prefixes)
+	local masked_ip
+
+	if prefix_len <= 24 then
+		return 0
+	end
+
+	-- For a prefix with length longer than 24, one tbl8
+	-- is needed according to the addtion description in
+	-- DPDK LPM library:
+	-- https://doc.dpdk.org/guides/prog_guide/lpm_lib.html#addition
+	masked_ip = lpmlib.ip_mask_addr(ip_addr, 24)
+	prefixes[masked_ip] = true
```

Comment:
Must test that `prefixes[masked_ip]` is not set before we set it; otherwise, we always return 1 for duplicates.

The same comment applies to the IPv6 version.